RawSpeed/TiffIFD.cpp

```diff
@@ -61,6 +61,11 @@ TiffIFD::TiffIFD(FileMap* f, uint32 offset, uint32 _depth) {
       t = new TiffEntry(f, entry_offset, offset);
     } catch (IOException) { // Ignore unparsable entry
       continue;
+    } catch (...) {
+      // make sure all already allocated mEntries are freed again
+      // will be gone after cleanup using unique_ptr containers
+      this->~TiffIFD();
```

Comment:
The not leaking for duplicate tags was a few lines below. This was for not leaking already parsed tags when a later tag (usually a MakerNote) would throw. I believe this particular leak was introduced by b453d047dfcbb74378b9caabe0258fffd315da4a because before there was no such exception thrown.